pkg/antctl/raw/multicluster/get/clusterset.go

```diff
@@ -0,0 +1,161 @@
+}
+
+var optionsClusterSet *clusterSetOptions
+
+var clusterSetExamples = strings.Trim(`
+Gel all ClusterSets in default Namesapce
+$ antctl mc get clusterset
+Get all ClusterSets in all Namespaces
+$ antctl mc get clusterset -A
+Get all ClusterSets in the specified Namespace
+$ antctl mc get clusterset -n <NAMESPACE>
+Get all ClusterSets and print them in JSON format
+$ antctl mc get clusterset -o json
+Get the specified ClusterSet
+$ antctl mc get clusterset <CLUSTERSETID>
+`, "\n")
+
+func (o *clusterSetOptions) validateAndComplete() {
+	if o.allNamespaces {
+		o.namespace = metav1.NamespaceAll
+		return
+	}
+	if o.namespace == "" {
+		o.namespace = metav1.NamespaceDefault
+		return
+	}
+}
+
+func NewClusterSetCommand() *cobra.Command {
+	cmdClusterSet = &cobra.Command{
+		Use: "clusterset",
+		Aliases: []string{
+			"clustersets",
+		},
+		Short:   "Print Multi-cluster ClusterSets",
+		Args:    cobra.MaximumNArgs(1),
+		Example: clusterSetExamples,
+		RunE:    runEClusterSet,
+	}
+	o := &clusterSetOptions{}
+	optionsClusterSet = o
+	cmdClusterSet.Flags().StringVarP(&o.namespace, "namespace", "n", "", "Namespace of ClusterSets")
+	cmdClusterSet.Flags().StringVarP(&o.outputFormat, "output", "o", "", "Output format. Supported formats: json|yaml")
+	cmdClusterSet.Flags().BoolVarP(&o.allNamespaces, "all-namespaces", "A", false, "If present, list ClusterSets across all namespaces")
+
+	return cmdClusterSet
+}
+
+func runEClusterSet(cmd *cobra.Command, args []string) error {
+	optionsClusterSet.validateAndComplete()
+
+	kubeconfig, err := raw.ResolveKubeconfig(cmd)
+	if err != nil {
+		return err
+	}
+
+	argsNum := len(args)
+	singleResource := false
+	if argsNum > 0 {
+		singleResource = true
+	}
+
+	k8sClient, err := raw.AddSchemeToClientCfg(kubeconfig)
+	if err != nil {
+		return err
+	}
+
+	var clusterSets []multiclusterv1alpha1.ClusterSet
+	if singleResource {
+		clusterSetName := args[0]
+		clusterSet := multiclusterv1alpha1.ClusterSet{}
+		err = k8sClient.Get(context.TODO(), types.NamespacedName{
+			Namespace: optionsClusterSet.namespace,
+			Name:      clusterSetName,
+		}, &clusterSet)
+		if err != nil {
+			return err
+		}
+		gvks, unversioned, err := k8sClient.Scheme().ObjectKinds(&clusterSet)
+		if err != nil {
+			return err
+		}
+		if !unversioned && len(gvks) == 1 {
+			clusterSet.SetGroupVersionKind(gvks[0])
+		}
+		clusterSets = append(clusterSets, clusterSet)
+	} else {
+		clusterSetList := &multiclusterv1alpha1.ClusterSetList{}
+		err = k8sClient.List(context.TODO(), clusterSetList, &client.ListOptions{Namespace: optionsClusterSet.namespace})
+		if err != nil {
+			return err
+		}
+		clusterSets = clusterSetList.Items
+	}
+
+	if len(clusterSets) == 0 {
+		if optionsClusterSet.namespace != "" {
+			fmt.Fprintf(cmd.ErrOrStderr(), "No resource found in Namespace %s\n", optionsClusterSet.namespace)
+		} else {
+			fmt.Fprintln(cmd.ErrOrStderr(), "No resources found")
+		}
+		return nil
+	}
+
+	switch optionsClusterSet.outputFormat {
```

Comment:
Maybe it could reuse what `output` function has done and move some logic to `clusterset.Transform` so this file doesn't have format related code, e.g.
```
return output(clusterSets, singleResource, optionsResourceExport.outputFormat, resourceexport.Transform)
``` 

```
func Transform(r interface{}, single bool) (interface{}, error) {
	if single {
		return objectTransform(r)
	}
	return listTransform(r)
}

func listTransform(l interface{}) (interface{}, error) {
	clusterSets := l.([]multiclusterv1alpha1.ClusterSet)
        ...
}

func objectTransform(o interface{}) (interface{}, error) {
	clusterSet := o.(multiclusterv1alpha1.ClusterSet)
        ...
}

func clusterTransform(clusterSet multiclusterv1alpha1.ClusterSet,
        status multiclusterv1alpha1.ClusterStatus,
	condition multiclusterv1alpha1.ClusterCondition) (interface{}, error) {
        ...
}
```

Reply:
@tnqn Hi Quan, I tried some files, but the blockers for me is here:  we don't want to change the function `TableOutputForGetCommands`, but we want to output the table output result one by one, so the "tableHeader" will always output once, but jianjun wanted to output them multiple times here

Reply:
Got it.